[ADD] edi_stock_oca

albariera opened the pull request:
This module intends to create a base to be extended by local edi rules
for stock.

albariera committed:
[ADD] edi_stock_oca

simahawk commented on edi_stock_oca/models/stock_picking.py:
```diff
@@ -0,0 +1,26 @@
+
+    _inherit = ["stock.picking", "edi.exchange.consumer.mixin"]
+
+    disable_edi_auto = fields.Boolean(
```
Question: would be interesting to have this on the consumer mixin?
Also, matter of taste maybe but I'd prefer to prefix such fields w/ `edi`. In this case I vote for `edi_auto_disabled`.

etobella replied:
Well, the field can be added on edi.exchange.mixin from edi_oca, but, at the end, we will allways need to inherit on the models in order to set the states and so on :thinking: 

I agree with the change of name and move to `edi`, but it is already being used from l10n-spain facturae (at least...) could we add a comment to rename it on 16?

simahawk replied:
IMO as this module is new this is the best time to change its name before is adopted by a lot of projects.

I propose this: 

1. change it here
2. for Spain localization, add `disable_edi_auto = fields.Boolean(related="edi_auto_disabled", readonly=False)`

Then you get rid of it when you want on Spain loc. Would it work for you?

If not, I won't keep it till v16. We are on v13 here, you can change this on v14, no?

etobella replied:
:laughing: I thought the same, but I didn't want to make it too hard :laughing:

Let's do this then

simahawk commented on edi_stock_oca/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,14 @@
+        _inherit = "base.event.listener"
+        _apply_on = ["stock.picking"]
+
+        def on_validate_stock_picking(self, picking):
```
```suggestion
        def on_validate(self, picking):
```
Seems redundant otherwise.

simahawk commented on edi_stock_oca/views/stock_picking.xml:
```diff
@@ -0,0 +1,37 @@
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form" />
+        <field name="arch" type="xml">
+            <group name="other_infos" position="after">
```
Wouldn't be better to have a tab for EDI?

simahawk replied:
ping :)

albariera replied:
Yes! Sorry, I thought I had already changed it. Done :)

simahawk reviewed (changes requested):
@SilvioC2C you might be interested in this

albariera committed:
[IMP] edi: move field edi_auto_disabled to edi.exchange.consumer.mixin

This field was defined at many models inheriting from edi.exchange.consumer.mixin. So it was thought that had more sense to have at the base model.

albariera committed:
[IMP] edi_account: change name field disable_edi_auto to edi_auto_disabled

simahawk commented on edi_account/models/account_move.py:
```diff
@@ -8,12 +8,13 @@ class AccountMove(models.Model):
     )
 
+    disable_edi_auto = fields.Boolean(related="edi_auto_disabled")
+    # TODO: remove this field at version 16.0
```
why not 14.0?

albariera replied:
Because edi_account is already migrated to 14.0 and 15.0. I was leaving it just for security reasons, maybe someone has already custom modules based on this module at those versions. Do you think it is better to remove it at 14.0?

simahawk replied:
IMO: yes. For existing installations you probably need a small migration script (the did here is ok).
What you can do: define a post install hook that can be used later for upgrades too (you can import an hook from the `migrations` folder but you cannot import vice versa).  Up to you... I don't have a full background on how and where you are using this already... you guys know better ;)

simahawk commented on edi_stock_oca/tests/test_edi.py:
```diff
@@ -0,0 +1,55 @@
+from odoo.addons.component.tests.common import SavepointComponentRegistryCase
+from odoo.addons.stock.tests.common import TestStockCommon
+
+_logger = logging.getLogger(__name__)
```
logger for?

albariera replied:
Removed 👍

simahawk commented:
@albariera pre-commit to be fixed. Also, try to keep commit msg shorter (move a long description to the commit long desc :wink: )

etobella commented:
The precommit error is solved by updating copier, but several issues were found with some other modules #562

olgamarcocb reviewed (approved):
LGTM :+1:

LoisRForgeFlow commented on edi_stock_oca/models/stock_picking.py:
```diff
@@ -0,0 +1,24 @@
+        },
+    )
+
+    def button_validate(self):
+        res = super().button_validate()
+        self._event("on_validate").notify(self)
+        return res
```
Wouldn't it be better to listen at `action_done`?, in button_validate there are several events that can pop before actually doing the picking (backorder wizard, no quantity done wizard, etc.).

albariera replied:
You are right. Changed! Thanks :)

etobella replied:
@LoisRForgeFlow Can you recheck it please :smile: ?

LoisRForgeFlow reviewed (approved):
LGTM, thanks

etobella commented:
/ocabot merge patch
